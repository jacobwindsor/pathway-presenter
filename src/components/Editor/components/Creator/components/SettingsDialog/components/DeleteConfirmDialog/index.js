import React from 'react';
import PropTypes from 'prop-types';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';

const DeleteConfirmDialog = props => {
    const actions = [
        <FlatButton
            label="Cancel"
            primary={true}
            onTouchTap={props.handleClose}
        />,
        <FlatButton
            label="Delete"
            primary={true}
            keyboardFocused={true}
            onTouchTap={() => { props.handleDelete(); props.handleClose(); }}
        />,
    ];

    return (
        <Dialog
            title="Delete Presentation"
            actions={actions}
            modal={false}
            onRequestClose={props.handleClose}
            open={props.isOpen}
        >
            Are you sure you want to delete this presentation?
        </Dialog>
    )
};

DeleteConfirmDialog.propTypes = {
    handleDelete: PropTypes.func.isRequired,
    handleClose: PropTypes.func.isRequired,
    isOpen: PropTypes.bool.isRequired,
};

export default DeleteConfirmDialog;