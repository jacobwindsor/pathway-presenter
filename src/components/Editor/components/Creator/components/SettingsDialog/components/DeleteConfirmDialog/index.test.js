import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DeleteConfirmDialog from './index';

describe('The DeleteConfirmDialog. Shows a warning message before deleting the presentation', () => {
  it('should render correctly', () => {
    const comp = shallow(
      <DeleteConfirmDialog
        handleDelete={() => null}
        handleClose={() => null}
        isOpen
      />
    );
    expect(toJson(comp)).toMatchSnapshot();
  });
});
