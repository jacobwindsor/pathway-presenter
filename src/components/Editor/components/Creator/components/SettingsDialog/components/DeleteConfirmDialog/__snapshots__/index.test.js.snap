// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The DeleteConfirmDialog. Shows a warning message before deleting the presentation should render correctly 1`] = `
<Dialog
  actions={
    Array [
      <FlatButton
        disabled={false}
        fullWidth={false}
        label="Cancel"
        labelPosition="after"
        labelStyle={Object {}}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        primary={true}
        secondary={false}
    />,
      <FlatButton
        disabled={false}
        fullWidth={false}
        keyboardFocused={true}
        label="Delete"
        labelPosition="after"
        labelStyle={Object {}}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        primary={true}
        secondary={false}
    />,
    ]
  }
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  modal={false}
  onRequestClose={[Function]}
  open={true}
  repositionOnUpdate={true}
  title="Delete Presentation"
>
  Are you sure you want to delete this presentation?
</Dialog>
`;
